kaiax/builder/impl/tx_pool.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2025 The Kaia Authors
+// This file is part of the Kaia library.
+//
+// The Kaia library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Kaia library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Kaia library. If not, see <http://www.gnu.org/licenses/>.
+
+package impl
+
+import (
+	"errors"
+	"time"
+
+	"github.com/kaiachain/kaia/blockchain/types"
+	"github.com/kaiachain/kaia/kaiax"
+)
+
+var _ kaiax.TxPoolModule = (*BuilderModule)(nil)
+
+func (b *BuilderModule) PreAddTx(tx *types.Transaction, local bool) error {
+	txTime, ok := b.knownTxs[tx.Hash()]
+	if ok && time.Since(txTime.time) < KnownTxTimeout {
+		return errors.New("Unable to add known bundle tx into tx pool during lock period")
+	}
+	for _, module := range b.Modules {
+		if module.IsBundleTx(tx) {
+			newTxTime := txAndTime{
+				tx:   tx,
+				time: time.Now(),
+			}
+			if ok {
+				newTxTime.time = txTime.time
+			}
```

Comment:
Why do we need to set the previous `time` to tx after `KnownTxTimeout`? I understood that this case is for the tx that passed `KnownTxTimeout` but not yet removed since `PreReset` hasn't been called. So after this, the tx will be eventually removed from `knownTxs` in `PreReset` and is it intended?

Reply:
Yes, that's right. 
If we use current time, there is possibility to keep to hold the tx because of keeping to updating time and keeping to receive the tx by nodes.

Reply:
https://github.com/ulbqb/kaia/blob/cdc412bbf6f6b3751432794e2adff6522f4e9ac1/kaiax/builder/impl/tx_pool_wrapping.go#L74-L92
Ok, but I changed the timing of when I set tx due to other issues. Should I remove timeout tx?

Reply:
I think https://github.com/kaiachain/kaia/pull/309#discussion_r2034844202 is enough. Thanks.